### PR TITLE
typescript-formatter: init at 7.2.2

### LINKED ISSUE #224320: Package request: nodePackages.tsfmt
**Project description**
Formatter for Typescript.

**Metadata**
* homepage URL: https://www.npmjs.com/package/typescript-formatter
* source URL: https://github.com/vvakame/typescript-formatter
* license: MIT
* platforms: unix, linux

### PR DESCRIPTION
**Pull request by ocfox**
###### Description of changes
close #224320

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 225280` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>typescript-formatter</li>
  </ul>
</details>

File: pkgs/development/tools/typescript-formatter/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      --prefix NODE_PATH : ${typescript}/lib/node_modules
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    description = "A TypeScript code formatter powered by TypeScript Compiler Service";
+    homepage = "https://github.com/vvakame/typescript-formatter";
+    maintainers = [ maintainers.ocfox ];
+    license = licenses.mit;
```
Comment (felschr) [inline pkgs/development/tools/typescript-formatter/default.nix:35]:
```suggestion
    changelog = "https://github.com/vvakame/typescript-formatter/blob/${version}/CHANGELOG.md";
    license = licenses.mit;
    mainProgram = "tsfmt";
```

**Review by felschr — CHANGES_REQUESTED** [review]
typescript-formatter hasn't been maintained for a while: last release was on 2018-06-01, last commit on `master` branch on 2018-07-14.
Is it really worth adding this package?

**Comment by ocfox** [comment]
Agreed, I also think that packages that are no longer maintained or unpopular should not be added. Maybe you can tell someone who needs this package (@hashelq) some suggestions (another package with the same functionality)?

**Pull request closed without merging**